test/optimist/parser_test.rb

```diff
@@ -779,6 +781,20 @@ def test_arguments_passed_through_block
     assert_equal @goat, boat
   end
 
+  ## test-only access reader method so that we dont have to
+  ## expose settings in the public API.
+  class Optimist::Parser
+    def get_settings_for_testing ; return @settings ;end
+  end
+
+  def test_two_arguments_passed_through_block
+    newp = Parser.new(:abcd => 123, :efgh => "other" ) do |i|
+    end
```

Comment:
This is interesting - I actually thought this should raise an exception.  (We can discuss separately, I won't hold up the PR for this)